app/main.py

```diff
@@ -1,3 +1,46 @@
-def shop_trip():
-    # write your code here
-    pass
+import json
+from app.car import Car
+from app.shop import Shop
+from app.customer import Customer
+
+
+def shop_trip() -> None:
+    with open("app/config.json") as config:
+        infos = json.load(config)
+
+    fuel_price = infos["FUEL_PRICE"]
+
+    for item in infos["customers"]:
+        customer = Customer(
+            item["name"],
+            item["product_cart"],
+            item["location"],
+            item["money"],
+        )
+        customer.customer_info()
+        cars = Car(item["car"]["brand"], item["car"]["fuel_consumption"])
+        shop_dict = {}
+        for shop in infos["shops"]:
+            current_shop = 0
+            shop = Shop(
+                shop["name"],
+                shop["location"],
+                shop["products"]
+            )
```

Comment:
Why you create shops inside loop?

Reply:
Move creating of shops before loop

Reply:
I create a shop inside the loop, to sort the cheapest shop for customer

Reply:
It is not correct solution. Because creating shops inside loop take time also. When you can create them just once

Reply:
I don`t know how to make else, give a hint, please

Reply:
You're creating shops for every customer and doing lots of redundant work - shops should be created once before looping through customers